v2/AdminCore.Services/DashboardService.cs

```diff
@@ -2,39 +2,173 @@
+        null,
+        x => x.EventDates,
+        x => x.Employee,
+        x => x.EventType,
+        x => x.EventStatus);
+      return _mapper.Map<IList<EventDto>>(events);
     }
 
     public IDictionary<string, List<EmployeeSnapshotDto>> GetEmployeeSnapshotsByEmployeeId(int employeeId)
     {
-      throw new NotImplementedException();
+      var contractList = DatabaseContext.ContractRepository.Get
+      (
+        contract => GetEmployeeSnapshotByEmployeeIdQuery(employeeId, contract, DateTime.Today),
+        null,
+        contract => contract.Team
+      );
+      return ConvertContractListToEmployeeSnapshotDtoMap(contractList);
     }
 
     public IDictionary<string, List<EmployeeSnapshotDto>> GetTeamSnapshotDashboardEvents()
     {
-      throw new NotImplementedException();
+      var contractList = DatabaseContext.ContractRepository.Get
+      (
+        contract => GetTeamSnapshotDashboardEventsQuery(contract, DateTime.Today),
+        null,
+        contract => contract.Team
+      );
+      return ConvertContractListToEmployeeSnapshotDtoMap(contractList);
     }
 
     public IList<EventMessageDto> GetEventMessagesByEventId(int eventId)
     {
-      throw new NotImplementedException();
+      var eventMessages = DatabaseContext.EventMessageRepository.Get(eventMessage => eventMessage.Event.EventId == eventId);
+      return _mapper.Map<IList<EventMessageDto>>(eventMessages);
     }
+
     public IList<EventDto> GetTeamDashboardEvents(int employeeId, DateTime date)
     {
-      throw new NotImplementedException();
+      var eventList = DatabaseContext.EventRepository.Get
+      (
+        evnt => (evnt.EmployeeId == employeeId || evnt.Employee.Contracts.Any(contract => contract.EmployeeId == employeeId && DateService.ContractIsActiveDuringDate(contract, date))) &&
+                DateService.EventContainsEventDatesThatHappenDuringMonth(evnt.EventDates, date),
+        null,
+        evnt => evnt.EventStatus,
+        evnt => evnt.EventType,
+        evnt => evnt.EventMessages
+      );
+      return _mapper.Map<IList<EventDto>>(eventList);
+    }
+    
+    public static bool EmployeeDashboardEventsQuery(int employeeId, DateTime date, Event evnt)
+    {
+      const int cancelled = (int) EventStatuses.Cancelled;
+      var startOfMonth = DateService.GetMonthStartDate(date);
+      var endOfMonth = DateService.GetMonthEndDate(date);
+
+      return (evnt.EmployeeId == employeeId) &&
+             (evnt.EventStatus.EventStatusId != cancelled) &&
+             DateService.EventContainsEventDatesThatHappenDuringMonth(evnt.EventDates, date) &&
+             EmployeeHasActiveContractsDuringMonth(evnt.Employee.Contracts, startOfMonth, endOfMonth);
+    }
+
+    public static bool GetEmployeeSnapshotByEmployeeIdQuery(int employeeId, Contract contract, DateTime date)
+    {
+      return contract.EmployeeId == employeeId &&
```

Comment:
Same for this and other returns

Reply:
GetEmployeeSnapshotByEmployeeIdQuery is the extracted method